crates/compiler/gen_llvm/src/llvm/build.rs

```diff
@@ -2884,14 +2884,17 @@ pub(crate) fn build_exp_stmt<'a, 'ctx>(
 
                         other_layout if other_layout.is_refcounted(layout_interner) => {
                             if value.is_pointer_value() {
-                                let value_ptr = match other_layout {
-                                    LayoutRepr::Union(union_layout)
-                                        if union_layout
-                                            .stores_tag_id_in_pointer(env.target_info) =>
-                                    {
-                                        tag_pointer_clear_tag_id(env, value.into_pointer_value())
+                                let clear_tag_id = match other_layout {
+                                    LayoutRepr::Union(union_layout) => {
+                                        union_layout.stores_tag_id_in_pointer(env.target_info)
                                     }
-                                    _ => value.into_pointer_value(),
+                                    _ => false,
+                                };
```

Comment:
`
let clear_tag_id = matches!(other_layout, LayoutRepr::Union(union_layout) if union_layout.stores_tag_id_in_pointer(env.target_info));
`

Reply:
given the size of the expression I don't think that is really much of an improvement here. also it expands right back to what we have right here